R/use_import_from.R

```diff
@@ -0,0 +1,67 @@
+#' Import a function from another package
+#'
+#' `use_import_from()` imports a function from another package by adding the
+#' roxygen2 `@importFrom` tag to the package-level documentation (which can be
+#' created with [`use_package_doc()`]). Importing a function from another
+#' package allows you to refer to it without a namespace (e.g., `fun()` instead
+#' of `package::fun()`). `use_import_from()` will also re-load the current
+#' package so that `fun` is loaded, as well.
+#'
+#' @param package Package name
+#' @param fun A vector of function names
+#' @param load Logical. Re-load with [`pkgload::load_all()`]?
+#'
+#' @return Invisibly, `TRUE` if the package document has changed, `FALSE` if
+#'   not.
+#' @export
+#'
+#' @examples
+#'
+#' \dontrun{
+#' use_import_from("usethis", "ui_todo")
+#' }
+#'
+use_import_from <- function(package, fun, load = is_interactive()) {
+  check_is_package("use_import_from()")
+  check_uses_roxygen("use_import_from()")
+
+  if (!check_has_package_doc()) {
+    return(invisible(FALSE))
+  }
+
+  use_dependency(package, "Imports")
+
+  fun <- gsub("\\(.*\\)", "", fun)
+  fun <- glue_collapse(fun, sep = " ")
+  changed <- roxygen_ns_append(glue("@importFrom {package} {fun}")) &&
+    roxygen_update_ns()
+
+  if (load) {
+    ui_done("Loading {project_name()}")
+    pkgload::load_all(quiet = TRUE)
+  }
+
+  invisible(changed)
+}
+
+check_has_package_doc <- function() {
+  if (has_package_doc()) {
+    return(invisible(TRUE))
+  }
+
+  if (!is_interactive()) {
+    return(invisible(FALSE))
+  }
+
+  add_pkg_doc <- ui_yeah("{ui_code('use_import_from()')} requires \\
```

Comment:
Can you please follow https://style.tidyverse.org/syntax.html#long-lines here?

Reply:
@jennybc might have other suggestions on how we usually format long strings in usethis

Reply:
I have developed a certain internal style for the `ui_*()` functions (re: line breaks in source vs. line breaks on the Console) that probably horrifies @Hadley.

But I have actually reached this convention through some thought and experimentation. So you can look around in general, but here are some `ui_yeah()` examples:

https://github.com/r-lib/usethis/blob/db18e70c718b476bd074b5893e26aeb6e7427a0a/R/package.R#L96-L101

https://github.com/r-lib/usethis/blob/db18e70c718b476bd074b5893e26aeb6e7427a0a/R/pkgdown.R#L121-L127

Reply:
:o